src/main/java/com/meta/cp4m/Service.java

```diff
@@ -29,7 +29,7 @@
 public class Service<T extends Message> {
 
   private static final Logger LOGGER = LoggerFactory.getLogger(Service.class);
-  private final ExecutorService executorService = Executors.newCachedThreadPool();
+  private final ExecutorService executorService = Executors.newVirtualThreadPerTaskExecutor();
```

Comment:
this is irrelevant to the rest of the PR I've just been meaning to introduce this for awhile and I kept forgetting. Should substantially lower resource utilization in high traffic situations